src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -63,14 +82,56 @@ const nonPreviewCards: string[] = [
 export class HuiCardPicker extends LitElement {
   @property() public hass?: HomeAssistant;
 
+  @property() private _cards: Card[] = [];
+
   public lovelace?: LovelaceConfig;
 
   public cardPicked?: (cardConf: LovelaceCardConfig) => void;
 
+  private _filter?: string;
+
   private _unusedEntities?: string[];
 
   private _usedEntities?: string[];
 
+  private _renderCards = memoizeOne((cards: Card[]): CardElement[] => {
+    return cards.map((card: Card) => {
+      return {
+        card: card,
+        element: html` ${until(
+          this._renderCardElement(card),
```

Comment:
I was thinking about adding the outcome of `this._renderCardElement(card)` to `this._cards`. We now still do all the work when the outcome of the search changes.

Reply:
:ok_hand:

Reply:
OK hows this?

Reply:
perfect 👍